File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        HashMap<Character,Integer> h=new  HashMap<>();
        for(int i=0;i<s.length();i++){
            h.put(s.charAt(i),h.getOrDefault(s.charAt(i),0)+1);
        }
        for(int i=0;i<s.length();i++){
            if(h.get(s.charAt(i))==1){
                return i;
            }
        }
        return -1;
        
    }
}

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        m={}
        for i in s:
            if i not in m:
                m[i]=1
            else:
                m[i]+=1
        for k,v in m.items():
            if v==1:
                return s.index(k)
        return -1
        